When parallel make it will fail with multi copy, see
http://bugzilla.pokylinux.org/show_bug.cgi?id=1202

Upstream-Status: Pending

Signed-off-by: Kang Kai <kai.kang@windriver.com>
Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>

RP: Extended || true to all CP_ operations, they all can race e.g.:
| cp -f ./obj/gconfxx.h ./obj/gconfig.h
| cp: cannot create regular file `./obj/gconfig.h': File exists

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>

Rebase the patch to ghostscript-9.15
Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>

diff --git a/base/expat.mak b/base/expat.mak
index 4ee9c8c..2e16a9d 100644
--- a/base/expat.mak
+++ b/base/expat.mak
@@ -82,7 +82,7 @@ $(EXPATOBJ)xmltok.$(OBJ) : $(EXPATSRC)xmltok.c $(expat_xmltok_hdrs)
 # Copy the target definition we want
 $(EXPATGEN)expat.dev : $(TOP_MAKEFILES) $(EXPAT_MAK) \
  $(EXPATGEN)expat_$(SHARE_EXPAT).dev
-	$(CP_) $(EXPATGEN)expat_$(SHARE_EXPAT).dev $(EXPATGEN)expat.dev
+	$(CP_) $(EXPATGEN)expat_$(SHARE_EXPAT).dev $(EXPATGEN)expat.dev || true
 
 # Define the compiled in target
 $(EXPATGEN)expat_0.dev : $(EXPAT_MAK) $(ECHOGS_XE) $(expat_)
diff --git a/base/freetype.mak b/base/freetype.mak
index 16f2b95..409f75b 100644
--- a/base/freetype.mak
+++ b/base/freetype.mak
@@ -195,7 +195,7 @@ ft_winfonts=$(FTOBJ)winfnt.$(OBJ)
 
 # instantiate the requested build option (shared or compiled in)
 $(FTGEN)freetype.dev : $(TOP_MAKEFILES) $(FTGEN)freetype_$(SHARE_FT).dev
-	$(CP_) $(FTGEN)freetype_$(SHARE_FT).dev $(FTGEN)freetype.dev
+	$(CP_) $(FTGEN)freetype_$(SHARE_FT).dev $(FTGEN)freetype.dev || true
 
 # Define the shared version.
 $(FTGEN)freetype_1.dev : $(TOP_MAKEFILES) $(FT_MAK) $(ECHOGS_XE)
diff --git a/base/gs.mak b/base/gs.mak
index 3fc67df..1fc3e26 100644
--- a/base/gs.mak
+++ b/base/gs.mak
@@ -434,7 +434,7 @@ $(gconfxx_h) : $(ld_tr)
 
 $(gconfig_h) : $(gconfxx_h)
 	$(RM_) $(gconfig_h)
-	$(CP_) $(gconfxx_h) $(gconfig_h)
+	$(CP_) $(gconfxx_h) $(gconfig_h) || true
 	
 # The line above is an empty command; don't delete.
 
diff --git a/base/ijs.mak b/base/ijs.mak
index bc6d549..0139e1c 100644
--- a/base/ijs.mak
+++ b/base/ijs.mak
@@ -66,7 +66,7 @@ $(IJSGEN)ijslib_1.dev : $(TOP_MAKEFILES) $(IJS_MAK) $(ECHOGS_XE)
 
 
 $(IJSGEN)ijslib.dev : $(TOP_MAKEFILES) $(IJS_MAK) $(IJSGEN)ijslib_$(SHARE_IJS).dev
-	$(CP_) $(IJSGEN)ijslib_$(SHARE_IJS).dev $(IJSGEN)ijslib.dev
+	$(CP_) $(IJSGEN)ijslib_$(SHARE_IJS).dev $(IJSGEN)ijslib.dev || true
 
 
 ijs_h=$(IJSSRC)ijs.h
diff --git a/base/jbig2.mak b/base/jbig2.mak
index bb74630..44617a0 100644
--- a/base/jbig2.mak
+++ b/base/jbig2.mak
@@ -98,7 +98,7 @@ JBIG2O_=$(O_)$(JBIG2OBJ)
 
 # switch in the version of libjbig2.dev we're actually using
 $(JBIG2GEN)jbig2dec.dev : $(TOP_MAKEFILES) $(JBIG2GEN)jbig2dec_$(SHARE_JBIG2).dev
-	$(CP_) $(JBIG2GEN)jbig2dec_$(SHARE_JBIG2).dev $(JBIG2GEN)jbig2dec.dev
+	$(CP_) $(JBIG2GEN)jbig2dec_$(SHARE_JBIG2).dev $(JBIG2GEN)jbig2dec.dev || true
 
 # dev file for shared (separately built) jbig2dec library
 $(JBIG2GEN)jbig2dec_1.dev : $(TOP_MAKEFILES) $(JBIG2_MAK) $(ECHOGS_XE)
diff --git a/base/jpeg.mak b/base/jpeg.mak
index e300a04..9fdad5a 100644
--- a/base/jpeg.mak
+++ b/base/jpeg.mak
@@ -96,7 +96,7 @@ jconfig_h=$(GLGEN)jconfig.h
 jmorecfg_h=$(GLGEN)jmorecfg.h
 
 $(GLGEN)jconfig_.h : $(GLGEN)jconfig$(SHARE_JPEG).h $(MAKEFILE) $(MAKEDIRS)
-	$(CP_) $(GLGEN)jconfig$(SHARE_JPEG).h $(GLGEN)jconfig_.h
+	$(CP_) $(GLGEN)jconfig$(SHARE_JPEG).h $(GLGEN)jconfig_.h || true
 
 $(GLGEN)jconfig0.h : $(ECHOGS_XE) $(GLSRC)gsjconf.h $(stdpre_h) $(MAKEFILE)\
  $(MAKEDIRS)
@@ -108,13 +108,13 @@ $(GLGEN)jconfig1.h : $(ECHOGS_XE) $(JPEG_MAK) $(MAKEDIRS)
 	$(RMN_) $(GLGEN)jconfig0.h $(GLGEN)jconfig.h
 
 $(GLGEN)jconfig.h : $(GLGEN)jconfig0.h $(MAKEDIRS)
-	$(CP_) $(GLGEN)jconfig0.h $(GLGEN)jconfig.h
+	$(CP_) $(GLGEN)jconfig0.h $(GLGEN)jconfig.h || true
 
 $(GLGEN)jmorecf_.h : $(GLGEN)jmorecf$(SHARE_JPEG).h $(MAKEFILE) $(MAKEDIRS)
-	$(CP_) $(GLGEN)jmorecf$(SHARE_JPEG).h $(GLGEN)jmorecf_.h
+	$(CP_) $(GLGEN)jmorecf$(SHARE_JPEG).h $(GLGEN)jmorecf_.h || true
 
 $(GLGEN)jmorecf0.h : $(GLSRC)gsjmorec.h $(GLGEN)jmcorig.h $(MAKEDIRS)
-	$(CP_) $(GLSRC)gsjmorec.h $(GLGEN)jmorecf0.h
+	$(CP_) $(GLSRC)gsjmorec.h $(GLGEN)jmorecf0.h || true
 	$(RM_) $(GLGEN)jmorecf1.h
 
 $(GLGEN)jmorecf1.h : $(ECHOGS_XE) $(JPEG_MAK) $(MAKEDIRS)
@@ -122,10 +122,10 @@ $(GLGEN)jmorecf1.h : $(ECHOGS_XE) $(JPEG_MAK) $(MAKEDIRS)
 	$(RMN_) $(GLGEN)jmorecf0.h $(GLGEN)jmorecfg.h
 
 $(GLGEN)jmorecfg.h : $(GLGEN)jmorecf0.h $(MAKEDIRS)
-	$(CP_) $(GLGEN)jmorecf0.h $(GLGEN)jmorecfg.h
+	$(CP_) $(GLGEN)jmorecf0.h $(GLGEN)jmorecfg.h || true
 
 $(GLGEN)jmcorig.h : $(JSRC)jmorecfg.h $(MAKEDIRS)
-	$(CP_) $(JSRC)jmorecfg.h $(GLGEN)jmcorig.h
+	$(CP_) $(JSRC)jmorecfg.h $(GLGEN)jmcorig.h || true
 
 # Contrary to what some portability bigots assert as fact, C compilers are
 # not consistent about where they start searching for #included files:
@@ -141,23 +141,23 @@ $(GLGEN)jmcorig.h : $(JSRC)jmorecfg.h $(MAKEDIRS)
 JHCOPY=$(GLGEN)jinclude.h $(GLGEN)jpeglib.h
 
 $(GLGEN)jinclude.h : $(JSRC)jinclude.h $(MAKEDIRS)
-	$(CP_) $(JSRC)jinclude.h $(GLGEN)jinclude.h
+	$(CP_) $(JSRC)jinclude.h $(GLGEN)jinclude.h || true
 
 # jpeglib_.h doesn't really depend on jconfig.h or jmcorig.h,
 # but we choose to put the dependencies here rather than in the
 # definition of jpeglib__h.
 $(GLGEN)jpeglib_.h : $(GLGEN)jpeglib$(SHARE_JPEG).h $(MAKEDIRS)
-	$(CP_) $(GLGEN)jpeglib$(SHARE_JPEG).h $(GLGEN)jpeglib_.h
+	$(CP_) $(GLGEN)jpeglib$(SHARE_JPEG).h $(GLGEN)jpeglib_.h || true
 
 $(GLGEN)jpeglib0.h : $(JSRC)jpeglib.h $(jconfig_h) $(jmorecfg_h) $(MAKEDIRS)
-	$(CP_) $(JSRC)jpeglib.h $(GLGEN)jpeglib0.h
+	$(CP_) $(JSRC)jpeglib.h $(GLGEN)jpeglib0.h || true
 
 $(GLGEN)jpeglib1.h : $(ECHOGS_XE) $(JPEG_MAK) $(MAKEDIRS)
 	$(EXP)$(ECHOGS_XE) -w $(GLGEN)jpeglib1.h -x 23 include -x 203c jpeglib.h -x 3e
 
 # We also need jpeglib.h for #includes in the library itself.
 $(GLGEN)jpeglib.h : $(JSRC)jpeglib.h $(MAKEDIRS)
-	$(CP_) $(JSRC)jpeglib.h $(GLGEN)jpeglib.h
+	$(CP_) $(JSRC)jpeglib.h $(GLGEN)jpeglib.h || true
 
 # In order to avoid having to keep the dependency lists for the IJG code
 # accurate, we simply make all of them depend on the only files that
@@ -181,40 +181,40 @@ $(JGEN)jpegc0.dev : $(JPEG_MAK) $(ECHOGS_XE) $(jpegc0_)
 	$(SETMOD) $(JGEN)jpegc0 $(jpegc0_)
 
 $(JOBJ)jcomapi.$(OBJ) : $(JSRC)jcomapi.c $(JDEP)
-	$(CP_) $(JSRC)jcomapi.c $(GLGEN)jcomapi.c
+	$(CP_) $(JSRC)jcomapi.c $(GLGEN)jcomapi.c || true
 	$(JCC) $(JO_)jcomapi.$(OBJ) $(C_) $(GLGEN)jcomapi.c
 	$(RM_) $(GLGEN)jcomapi.c
 
 $(JOBJ)jutils.$(OBJ) : $(JSRC)jutils.c $(JDEP)
-	$(CP_) $(JSRC)jutils.c $(GLGEN)jutils.c
+	$(CP_) $(JSRC)jutils.c $(GLGEN)jutils.c || true
 	$(JCC) $(JO_)jutils.$(OBJ) $(C_) $(GLGEN)jutils.c
 	$(RM_) $(GLGEN)jutils.c
 
 $(JOBJ)jmemmgr.$(OBJ) : $(JSRC)jmemmgr.c $(JDEP)
-	$(CP_) $(JSRC)jmemmgr.c $(GLGEN)jmemmgr.c
+	$(CP_) $(JSRC)jmemmgr.c $(GLGEN)jmemmgr.c || true
 	$(JCC) $(JO_)jmemmgr.$(OBJ) $(C_) $(GLGEN)jmemmgr.c
 	$(RM_) $(GLGEN)jmemmgr.c
 
 $(JOBJ)jerror.$(OBJ) : $(JSRC)jerror.c $(JDEP)
-	$(CP_) $(JSRC)jerror.c $(GLGEN)jerror.c
+	$(CP_) $(JSRC)jerror.c $(GLGEN)jerror.c || true
 	$(JCC) $(JO_)jerror.$(OBJ) $(C_) $(GLGEN)jerror.c
 	$(RM_) $(GLGEN)jerror.c
 
 $(JOBJ)jaricom.$(OBJ) : $(JSRC)jaricom.c $(JDEP)
-	$(CP_) $(JSRC)jaricom.c $(GLGEN)jaricom.c
+	$(CP_) $(JSRC)jaricom.c $(GLGEN)jaricom.c || true
 	$(JCC) $(JO_)jaricom.$(OBJ) $(C_) $(GLGEN)jaricom.c
 	$(RM_) $(GLGEN)jaricom.c
 
 # Encoding (compression) code.
 
 $(JGEN)jpege.dev : $(TOP_MAKEFILES) $(JPEG_MAK) $(JGEN)jpege_$(SHARE_JPEG).dev
-	$(CP_) $(JGEN)jpege_$(SHARE_JPEG).dev $(JGEN)jpege.dev
+	$(CP_) $(JGEN)jpege_$(SHARE_JPEG).dev $(JGEN)jpege.dev || true
 
 $(JGEN)jpege_1.dev : $(TOP_MAKEFILES) $(JPEG_MAK) $(ECHOGS_XE)
 	$(SETMOD) $(JGEN)jpege_1 -lib $(JPEG_NAME)
 
 $(JGEN)jpege_0.dev : $(TOP_MAKEFILES) $(JPEG_MAK) $(JGEN)jpege6.dev
-	$(CP_) $(JGEN)jpege6.dev $(JGEN)jpege_0.dev
+	$(CP_) $(JGEN)jpege6.dev $(JGEN)jpege_0.dev || true
 
 jpege6=$(JOBJ)jcapimin.$(OBJ) $(JOBJ)jcapistd.$(OBJ) $(JOBJ)jcinit.$(OBJ)
 
@@ -230,90 +230,90 @@ $(JGEN)jpege6.dev : $(JPEG_MAK) $(ECHOGS_XE) $(JGEN)jpegc0.dev $(jpege6) $(jpege
 	$(ADDMOD) $(JGEN)jpege6 -obj $(jpege_3)
 
 $(JOBJ)jcapimin.$(OBJ) : $(JSRC)jcapimin.c $(JDEP)
-	$(CP_) $(JSRC)jcapimin.c $(GLGEN)jcapimin.c
+	$(CP_) $(JSRC)jcapimin.c $(GLGEN)jcapimin.c || true
 	$(JCC) $(JO_)jcapimin.$(OBJ) $(C_) $(GLGEN)jcapimin.c
 	$(RM_) $(GLGEN)jcapimin.c
 
 $(JOBJ)jcapistd.$(OBJ) : $(JSRC)jcapistd.c $(JDEP)
-	$(CP_) $(JSRC)jcapistd.c $(GLGEN)jcapistd.c
+	$(CP_) $(JSRC)jcapistd.c $(GLGEN)jcapistd.c || true
 	$(JCC) $(JO_)jcapistd.$(OBJ) $(C_) $(GLGEN)jcapistd.c
 	$(RM_) $(GLGEN)jcapistd.c
 
 $(JOBJ)jcinit.$(OBJ) : $(JSRC)jcinit.c $(JDEP)
-	$(CP_) $(JSRC)jcinit.c $(GLGEN)jcinit.c
+	$(CP_) $(JSRC)jcinit.c $(GLGEN)jcinit.c || true
 	$(JCC) $(JO_)jcinit.$(OBJ) $(C_) $(GLGEN)jcinit.c
 	$(RM_) $(GLGEN)jcinit.c
 
 $(JOBJ)jccoefct.$(OBJ) : $(JSRC)jccoefct.c $(JDEP)
-	$(CP_) $(JSRC)jccoefct.c $(GLGEN)jccoefct.c
+	$(CP_) $(JSRC)jccoefct.c $(GLGEN)jccoefct.c || true
 	$(JCC) $(JO_)jccoefct.$(OBJ) $(C_) $(GLGEN)jccoefct.c
 	$(RM_) $(GLGEN)jccoefct.c
 
 $(JOBJ)jccolor.$(OBJ) : $(JSRC)jccolor.c $(JDEP)
-	$(CP_) $(JSRC)jccolor.c $(GLGEN)jccolor.c
+	$(CP_) $(JSRC)jccolor.c $(GLGEN)jccolor.c || true
 	$(JCC) $(JO_)jccolor.$(OBJ) $(C_) $(GLGEN)jccolor.c
 	$(RM_) $(GLGEN)jccolor.c
 
 $(JOBJ)jcdctmgr.$(OBJ) : $(JSRC)jcdctmgr.c $(JDEP)
-	$(CP_) $(JSRC)jcdctmgr.c $(GLGEN)jcdctmgr.c
+	$(CP_) $(JSRC)jcdctmgr.c $(GLGEN)jcdctmgr.c || true
 	$(JCC) $(JO_)jcdctmgr.$(OBJ) $(C_) $(GLGEN)jcdctmgr.c
 	$(RM_) $(GLGEN)jcdctmgr.c
 
 $(JOBJ)jchuff.$(OBJ) : $(JSRC)jchuff.c $(JDEP)
-	$(CP_) $(JSRC)jchuff.c $(GLGEN)jchuff.c
+	$(CP_) $(JSRC)jchuff.c $(GLGEN)jchuff.c || true
 	$(JCC) $(JO_)jchuff.$(OBJ) $(C_) $(GLGEN)jchuff.c
 	$(RM_) $(GLGEN)jchuff.c
 
 $(JOBJ)jcmainct.$(OBJ) : $(JSRC)jcmainct.c $(JDEP)
-	$(CP_) $(JSRC)jcmainct.c $(GLGEN)jcmainct.c
+	$(CP_) $(JSRC)jcmainct.c $(GLGEN)jcmainct.c || true
 	$(JCC) $(JO_)jcmainct.$(OBJ) $(C_) $(GLGEN)jcmainct.c
 	$(RM_) $(GLGEN)jcmainct.c
 
 $(JOBJ)jcmarker.$(OBJ) : $(JSRC)jcmarker.c $(JDEP)
-	$(CP_) $(JSRC)jcmarker.c $(GLGEN)jcmarker.c
+	$(CP_) $(JSRC)jcmarker.c $(GLGEN)jcmarker.c || true
 	$(JCC) $(JO_)jcmarker.$(OBJ) $(C_) $(GLGEN)jcmarker.c
 	$(RM_) $(GLGEN)jcmarker.c
 
 $(JOBJ)jcmaster.$(OBJ) : $(JSRC)jcmaster.c $(JDEP)
-	$(CP_) $(JSRC)jcmaster.c $(GLGEN)jcmaster.c
+	$(CP_) $(JSRC)jcmaster.c $(GLGEN)jcmaster.c || true
 	$(JCC) $(JO_)jcmaster.$(OBJ) $(C_) $(GLGEN)jcmaster.c
 	$(RM_) $(GLGEN)jcmaster.c
 
 $(JOBJ)jcparam.$(OBJ) : $(JSRC)jcparam.c $(JDEP)
-	$(CP_) $(JSRC)jcparam.c $(GLGEN)jcparam.c
+	$(CP_) $(JSRC)jcparam.c $(GLGEN)jcparam.c || true
 	$(JCC) $(JO_)jcparam.$(OBJ) $(C_) $(GLGEN)jcparam.c
 	$(RM_) $(GLGEN)jcparam.c
 
 $(JOBJ)jcprepct.$(OBJ) : $(JSRC)jcprepct.c $(JDEP)
-	$(CP_) $(JSRC)jcprepct.c $(GLGEN)jcprepct.c
+	$(CP_) $(JSRC)jcprepct.c $(GLGEN)jcprepct.c || true
 	$(JCC) $(JO_)jcprepct.$(OBJ) $(C_) $(GLGEN)jcprepct.c
 	$(RM_) $(GLGEN)jcprepct.c
 
 $(JOBJ)jcsample.$(OBJ) : $(JSRC)jcsample.c $(JDEP)
-	$(CP_) $(JSRC)jcsample.c $(GLGEN)jcsample.c
+	$(CP_) $(JSRC)jcsample.c $(GLGEN)jcsample.c || true
 	$(JCC) $(JO_)jcsample.$(OBJ) $(C_) $(GLGEN)jcsample.c
 	$(RM_) $(GLGEN)jcsample.c
 
 $(JOBJ)jfdctint.$(OBJ) : $(JSRC)jfdctint.c $(JDEP)
-	$(CP_) $(JSRC)jfdctint.c $(GLGEN)jfdctint.c
+	$(CP_) $(JSRC)jfdctint.c $(GLGEN)jfdctint.c || true
 	$(JCC) $(JO_)jfdctint.$(OBJ) $(C_) $(GLGEN)jfdctint.c
 	$(RM_) $(GLGEN)jfdctint.c
 
 $(JOBJ)jcarith.$(OBJ) : $(JSRC)jcarith.c $(JDEP)
-	$(CP_) $(JSRC)jcarith.c $(GLGEN)jcarith.c
+	$(CP_) $(JSRC)jcarith.c $(GLGEN)jcarith.c || true
 	$(JCC) $(JO_)jcarith.$(OBJ) $(C_) $(GLGEN)jcarith.c
 	$(RM_) $(GLGEN)jcarith.c
 
 # Decompression code
 
 $(JGEN)jpegd.dev : $(TOP_MAKEFILES) $(JPEG_MAK) $(JGEN)jpegd_$(SHARE_JPEG).dev
-	$(CP_) $(JGEN)jpegd_$(SHARE_JPEG).dev $(JGEN)jpegd.dev
+	$(CP_) $(JGEN)jpegd_$(SHARE_JPEG).dev $(JGEN)jpegd.dev || true
 
 $(JGEN)jpegd_1.dev : $(TOP_MAKEFILES) $(JPEG_MAK) $(ECHOGS_XE)
 	$(SETMOD) $(JGEN)jpegd_1 -lib $(JPEG_NAME)
 
 $(JGEN)jpegd_0.dev : $(TOP_MAKEFILES) $(JPEG_MAK) $(JGEN)jpegd6.dev
-	$(CP_) $(JGEN)jpegd6.dev $(JGEN)jpegd_0.dev
+	$(CP_) $(JGEN)jpegd6.dev $(JGEN)jpegd_0.dev || true
 
 jpegd6=$(JOBJ)jdapimin.$(OBJ) $(JOBJ)jdapistd.$(OBJ) $(JOBJ)jdinput.$(OBJ) $(JOBJ)jdhuff.$(OBJ)
 
@@ -329,76 +329,76 @@ $(JGEN)jpegd6.dev : $(JPEG_MAK) $(ECHOGS_XE) $(JGEN)jpegc0.dev $(jpegd6) $(jpegd
 	$(ADDMOD) $(JGEN)jpegd6 -obj $(jpegd_3)
 
 $(JOBJ)jdapimin.$(OBJ) : $(JSRC)jdapimin.c $(JDEP)
-	$(CP_) $(JSRC)jdapimin.c $(GLGEN)jdapimin.c
+	$(CP_) $(JSRC)jdapimin.c $(GLGEN)jdapimin.c || true
 	$(JCC) $(JO_)jdapimin.$(OBJ) $(C_) $(GLGEN)jdapimin.c
 	$(RM_) $(GLGEN)jdapimin.c
 
 $(JOBJ)jdapistd.$(OBJ) : $(JSRC)jdapistd.c $(JDEP)
-	$(CP_) $(JSRC)jdapistd.c $(GLGEN)jdapistd.c
+	$(CP_) $(JSRC)jdapistd.c $(GLGEN)jdapistd.c || true
 	$(JCC) $(JO_)jdapistd.$(OBJ) $(C_) $(GLGEN)jdapistd.c
 	$(RM_) $(GLGEN)jdapistd.c
 
 $(JOBJ)jdcoefct.$(OBJ) : $(JSRC)jdcoefct.c $(JDEP)
-	$(CP_) $(JSRC)jdcoefct.c $(GLGEN)jdcoefct.c
+	$(CP_) $(JSRC)jdcoefct.c $(GLGEN)jdcoefct.c || true
 	$(JCC) $(JO_)jdcoefct.$(OBJ) $(C_) $(GLGEN)jdcoefct.c
 	$(RM_) $(GLGEN)jdcoefct.c
 
 $(JOBJ)jdcolor.$(OBJ) : $(JSRC)jdcolor.c $(JDEP)
-	$(CP_) $(JSRC)jdcolor.c $(GLGEN)jdcolor.c
+	$(CP_) $(JSRC)jdcolor.c $(GLGEN)jdcolor.c || true
 	$(JCC) $(JO_)jdcolor.$(OBJ) $(C_) $(GLGEN)jdcolor.c
 	$(RM_) $(GLGEN)jdcolor.c
 
 $(JOBJ)jddctmgr.$(OBJ) : $(JSRC)jddctmgr.c $(JDEP)
-	$(CP_) $(JSRC)jddctmgr.c $(GLGEN)jddctmgr.c
+	$(CP_) $(JSRC)jddctmgr.c $(GLGEN)jddctmgr.c || true
 	$(JCC) $(JO_)jddctmgr.$(OBJ) $(C_) $(GLGEN)jddctmgr.c
 	$(RM_) $(GLGEN)jddctmgr.c
 
 $(JOBJ)jdhuff.$(OBJ) : $(JSRC)jdhuff.c $(JDEP)
-	$(CP_) $(JSRC)jdhuff.c $(GLGEN)jdhuff.c
+	$(CP_) $(JSRC)jdhuff.c $(GLGEN)jdhuff.c || true
 	$(JCC) $(JO_)jdhuff.$(OBJ) $(C_) $(GLGEN)jdhuff.c
 	$(RM_) $(GLGEN)jdhuff.c
 
 $(JOBJ)jdinput.$(OBJ) : $(JSRC)jdinput.c $(JDEP)
-	$(CP_) $(JSRC)jdinput.c $(GLGEN)jdinput.c
+	$(CP_) $(JSRC)jdinput.c $(GLGEN)jdinput.c || true
 	$(JCC) $(JO_)jdinput.$(OBJ) $(C_) $(GLGEN)jdinput.c
 	$(RM_) $(GLGEN)jdinput.c
 
 $(JOBJ)jdmainct.$(OBJ) : $(JSRC)jdmainct.c $(JDEP)
-	$(CP_) $(JSRC)jdmainct.c $(GLGEN)jdmainct.c
+	$(CP_) $(JSRC)jdmainct.c $(GLGEN)jdmainct.c || true
 	$(JCC) $(JO_)jdmainct.$(OBJ) $(C_) $(GLGEN)jdmainct.c
 	$(RM_) $(GLGEN)jdmainct.c
 
 $(JOBJ)jdmarker.$(OBJ) : $(JSRC)jdmarker.c $(JDEP)
-	$(CP_) $(JSRC)jdmarker.c $(GLGEN)jdmarker.c
+	$(CP_) $(JSRC)jdmarker.c $(GLGEN)jdmarker.c || true
 	$(JCC) $(JO_)jdmarker.$(OBJ) $(C_) $(GLGEN)jdmarker.c
 	$(RM_) $(GLGEN)jdmarker.c
 
 $(JOBJ)jdmaster.$(OBJ) : $(JSRC)jdmaster.c $(JDEP)
-	$(CP_) $(JSRC)jdmaster.c $(GLGEN)jdmaster.c
+	$(CP_) $(JSRC)jdmaster.c $(GLGEN)jdmaster.c || true
 	$(JCC) $(JO_)jdmaster.$(OBJ) $(C_) $(GLGEN)jdmaster.c
 	$(RM_) $(GLGEN)jdmaster.c
 
 #$(JOBJ)jdhuff.$(OBJ) : $(JSRC)jdhuff.c $(JDEP)
-#	$(CP_) $(JSRC)jdhuff.c $(GLGEN)jdhuff.c
+#	$(CP_) $(JSRC)jdhuff.c $(GLGEN)jdhuff.c || true
 #	$(JCC) $(JO_)jdhuff.$(OBJ) $(C_) $(GLGEN)jdhuff.c
 #	$(RM_) $(GLGEN)jdhuff.c
 
 $(JOBJ)jdpostct.$(OBJ) : $(JSRC)jdpostct.c $(JDEP)
-	$(CP_) $(JSRC)jdpostct.c $(GLGEN)jdpostct.c
+	$(CP_) $(JSRC)jdpostct.c $(GLGEN)jdpostct.c || true
 	$(JCC) $(JO_)jdpostct.$(OBJ) $(C_) $(GLGEN)jdpostct.c
 	$(RM_) $(GLGEN)jdpostct.c
 
 $(JOBJ)jdsample.$(OBJ) : $(JSRC)jdsample.c $(JDEP)
-	$(CP_) $(JSRC)jdsample.c $(GLGEN)jdsample.c
+	$(CP_) $(JSRC)jdsample.c $(GLGEN)jdsample.c || true
 	$(JCC) $(JO_)jdsample.$(OBJ) $(C_) $(GLGEN)jdsample.c
 	$(RM_) $(GLGEN)jdsample.c
 
 $(JOBJ)jidctint.$(OBJ) : $(JSRC)jidctint.c $(JDEP)
-	$(CP_) $(JSRC)jidctint.c $(GLGEN)jidctint.c
+	$(CP_) $(JSRC)jidctint.c $(GLGEN)jidctint.c || true
 	$(JCC) $(JO_)jidctint.$(OBJ) $(C_) $(GLGEN)jidctint.c
 	$(RM_) $(GLGEN)jidctint.c
 
 $(JOBJ)jdarith.$(OBJ) : $(JSRC)jdarith.c $(JDEP)
-	$(CP_) $(JSRC)jdarith.c $(GLGEN)jdarith.c
+	$(CP_) $(JSRC)jdarith.c $(GLGEN)jdarith.c || true
 	$(JCC) $(JO_)jdarith.$(OBJ) $(C_) $(GLGEN)jdarith.c
 	$(RM_) $(GLGEN)jdarith.c
diff --git a/base/jpegxr.mak b/base/jpegxr.mak
index 0f09a3a..36495dd 100644
--- a/base/jpegxr.mak
+++ b/base/jpegxr.mak
@@ -103,7 +103,7 @@ $(JPEGXR_OBJ)x_strip.$(OBJ) : $(JPEGXR_SRC)x_strip.c $(jpegxr_hdrs)
 # Copy the target definition we want
 $(JPEGXR_GEN)jpegxr.dev : $(TOP_MAKEFILES) $(JPEGXR_MAK) \
  $(JPEGXR_GEN)jpegxr_$(SHARE_JPEGXR).dev
-	$(CP_) $(JPEGXR_GEN)jpegxr_$(SHARE_JPEGXR).dev $(JPEGXR_GEN)jpegxr.dev
+	$(CP_) $(JPEGXR_GEN)jpegxr_$(SHARE_JPEGXR).dev $(JPEGXR_GEN)jpegxr.dev || true
 
 # Define the compiled in target
 $(JPEGXR_GEN)jpegxr_0.dev : $(JPEGXR_MAK) $(ECHOGS_XE) $(jpegxr_objs)
diff --git a/base/lcms.mak b/base/lcms.mak
index 18cfa09..269e7ff 100644
--- a/base/lcms.mak
+++ b/base/lcms.mak
@@ -84,7 +84,7 @@ LCMSO_=$(O_)$(LCMSOBJ)
 
 # switch in the version of lcms.dev we're actually using
 $(LCMSGEN)lcms.dev : $(TOP_MAKEFILES) $(LCMSGEN)lcms_$(SHARE_LCMS).dev
-	$(CP_) $(LCMSGEN)lcms_$(SHARE_LCMS).dev $(LCMSGEN)lcms.dev
+	$(CP_) $(LCMSGEN)lcms_$(SHARE_LCMS).dev $(LCMSGEN)lcms.dev || true
 
 # dev file for shared (separately built) lcms library
 $(LCMSGEN)lcms_1.dev : $(TOP_MAKEFILES) $(LCMS_MAK) $(ECHOGS_XE)
diff --git a/base/lcms2.mak b/base/lcms2.mak
index 52f750c..33a9c16 100644
--- a/base/lcms2.mak
+++ b/base/lcms2.mak
@@ -85,7 +85,7 @@ LCMS2O_=$(O_)$(LCMS2OBJ)
 
 # switch in the version of lcms2.dev we're actually using
 $(LCMS2GEN)lcms2.dev : $(TOP_MAKEFILES) $(LCMS2GEN)lcms2_$(SHARE_LCMS).dev
-	$(CP_) $(LCMS2GEN)lcms2_$(SHARE_LCMS).dev $(LCMS2GEN)lcms2.dev
+	$(CP_) $(LCMS2GEN)lcms2_$(SHARE_LCMS).dev $(LCMS2GEN)lcms2.dev || true
 
 # dev file for shared (separately built) lcms library
 $(LCMS2GEN)lcms2_1.dev : $(TOP_MAKEFILES) $(LCMS2_MAK) $(ECHOGS_XE)
diff --git a/base/lcups.mak b/base/lcups.mak
index 7504c44..7b9b979 100644
--- a/base/lcups.mak
+++ b/base/lcups.mak
@@ -116,7 +116,7 @@ libcups.config-clean :
 
 # instantiate the requested build option (shared or compiled in)
 $(LIBCUPSGEN)lcups.dev : $(TOP_MAKEFILES) $(LIBCUPSGEN)lcups_$(SHARE_LCUPS).dev
-	$(CP_) $(LIBCUPSGEN)lcups_$(SHARE_LCUPS).dev $(LIBCUPSGEN)lcups.dev
+	$(CP_) $(LIBCUPSGEN)lcups_$(SHARE_LCUPS).dev $(LIBCUPSGEN)lcups.dev || true
 
 # Define the shared version.
 $(LIBCUPSGEN)lcups_1.dev : $(TOP_MAKEFILES) $(LCUPS_MAK) $(ECHOGS_XE)
@@ -133,7 +133,7 @@ $(LIBCUPSGEN)lcups_0.dev : $(TOP_MAKEFILES) $(LCUPS_MAK) $(ECHOGS_XE) \
 # for simplicity we have every source file depend on all headers
 
 $(LIBCUPSGEN)$(D)cups$(D)config.h : $(LCUPSSRCDIR)$(D)libs$(D)config$(LCUPSBUILDTYPE).h
-	$(CP_) $(LCUPSSRCDIR)$(D)libs$(D)config$(LCUPSBUILDTYPE).h $(LIBCUPSGEN)$(D)cups$(D)config.h
+	$(CP_) $(LCUPSSRCDIR)$(D)libs$(D)config$(LCUPSBUILDTYPE).h $(LIBCUPSGEN)$(D)cups$(D)config.h || true
 
 $(LIBCUPSOBJ)adminutil.$(OBJ) : $(LIBCUPSSRC)adminutil.c $(LIBSCUPSHEADERS) $(LIBCUPSGEN)$(D)cups$(D)config.h
 	$(LCUPS_CC) $(LCUPSO_)adminutil.$(OBJ) $(C_) $(LIBCUPSSRC)adminutil.c
@@ -220,7 +220,7 @@ $(LIBCUPSOBJ)mark.$(OBJ) : $(LIBCUPSSRC)mark.c $(LIBSCUPSHEADERS)
 	$(LCUPS_CC) $(LCUPSO_)mark.$(OBJ) $(C_) $(LIBCUPSSRC)mark.c
 
 $(LIBCUPSOBJ)cups_md5.$(OBJ) : $(LIBCUPSSRC)md5.c $(LIBSCUPSHEADERS)
-	$(CP_) $(LIBCUPSSRC)md5.c $(LIBCUPSGEN)cups_md5.c
+	$(CP_) $(LIBCUPSSRC)md5.c $(LIBCUPSGEN)cups_md5.c || true
 	$(LCUPS_CC) $(LCUPSO_)cups_md5.$(OBJ) $(C_) $(LIBCUPSGEN)cups_md5.c
 
 $(LIBCUPSOBJ)md5passwd.$(OBJ) : $(LIBCUPSSRC)md5passwd.c $(LIBSCUPSHEADERS)
@@ -257,7 +257,7 @@ $(LIBCUPSOBJ)snmp.$(OBJ) : $(LIBCUPSSRC)snmp.c $(LIBSCUPSHEADERS)
 	$(LCUPS_CC) $(LCUPSO_)snmp.$(OBJ) $(C_) $(LIBCUPSSRC)snmp.c
 
 $(LIBCUPSOBJ)cups_snpf.$(OBJ) : $(LIBCUPSSRC)snprintf.c $(LIBSCUPSHEADERS)
-	$(CP_) $(LIBCUPSSRC)snprintf.c $(LIBCUPSGEN)cups_snpf.c
+	$(CP_) $(LIBCUPSSRC)snprintf.c $(LIBCUPSGEN)cups_snpf.c || true
 	$(LCUPS_CC) $(LCUPSO_)cups_snpf.$(OBJ) $(C_) $(LIBCUPSGEN)cups_snpf.c
 
 $(LIBCUPSOBJ)string.$(OBJ) : $(LIBCUPSSRC)string.c $(LIBSCUPSHEADERS)
@@ -279,5 +279,5 @@ $(LIBCUPSOBJ)thread.$(OBJ) : $(LIBCUPSSRC)thread.c $(LIBSCUPSHEADERS)
 	$(LCUPS_CC) $(LCUPSO_)thread.$(OBJ) $(C_) $(LIBCUPSSRC)thread.c
 
 $(LIBCUPSOBJ)cups_util.$(OBJ) : $(LIBCUPSSRC)util.c $(LIBSCUPSHEADERS)
-	$(CP_) $(LIBCUPSSRC)util.c $(LIBCUPSGEN)cups_util.c
+	$(CP_) $(LIBCUPSSRC)util.c $(LIBCUPSGEN)cups_util.c || true
 	$(LCUPS_CC) $(LCUPSO_)cups_util.$(OBJ) $(C_) $(LIBCUPSGEN)cups_util.c
diff --git a/base/lcupsi.mak b/base/lcupsi.mak
index da47da5..fa7d74c 100644
--- a/base/lcupsi.mak
+++ b/base/lcupsi.mak
@@ -59,7 +59,7 @@ libcupsi.config-clean :
 
 # instantiate the requested build option (shared or compiled in)
 $(LIBCUPSIGEN)lcupsi.dev : $(TOP_MAKEFILES) $(LIBCUPSIGEN)lcupsi_$(SHARE_LCUPSI).dev
-	$(CP_) $(LIBCUPSIGEN)lcupsi_$(SHARE_LCUPSI).dev $(LIBCUPSIGEN)lcupsi.dev
+	$(CP_) $(LIBCUPSIGEN)lcupsi_$(SHARE_LCUPSI).dev $(LIBCUPSIGEN)lcupsi.dev || true
 
 # Define the shared version.
 $(LIBCUPSIGEN)lcupsi_1.dev : $(TOP_MAKEFILES) $(LCUPSI_MAK) $(ECHOGS_XE)
diff --git a/base/ldf_jb2.mak b/base/ldf_jb2.mak
index 3f50892..97c76a0 100644
--- a/base/ldf_jb2.mak
+++ b/base/ldf_jb2.mak
@@ -223,7 +223,7 @@ ldf_jb2_HDRS=$(ldf_jb2_common_HDRS) $(ldf_jb2_compress_HDRS)
 
 # switch in the selected library .dev
 $(LDF_JB2_GEN)ldf_jb2.dev : $(TOP_MAKEFILES) $(LDF_JB2_MAK) $(LDF_JB2_GEN)ldf_jb2_$(SHARE_JBIG2).dev
-	$(CP_) $(LDF_JB2_GEN)ldf_jb2_$(SHARE_JBIG2).dev $(LDF_JB2_GEN)ldf_jb2.dev
+	$(CP_) $(LDF_JB2_GEN)ldf_jb2_$(SHARE_JBIG2).dev $(LDF_JB2_GEN)ldf_jb2.dev || true
 
 # external link .dev
 $(LDF_JB2_GEN)ldf_jb2_1.dev : $(TOP_MAKEFILES) $(LDF_JB2_MAK) $(ECHOGS_XE)
diff --git a/base/lib.mak b/base/lib.mak
index 90c9249..fb11aba 100644
--- a/base/lib.mak
+++ b/base/lib.mak
@@ -350,7 +350,7 @@ md5_=$(GLOBJ)md5.$(OBJ)
 $(GLOBJ)md5.$(OBJ) : $(GLSRC)md5.c $(AK) $(md5_h) $(std_h) $(MAKEDIRS) $(EXP)$(ECHOGS_XE)
 	$(EXP)$(ECHOGS_XE) -w $(GLGEN)md5.h -x 23 include -x 2022 memory_.h -x 22
 	$(EXP)$(ECHOGS_XE) -a $(GLGEN)md5.h -+R $(GLSRC)md5.h
-	$(CP_) $(GLSRC)md5.c $(GLGEN)md5.c
+	$(CP_) $(GLSRC)md5.c $(GLGEN)md5.c || true
 	$(GLCC) $(GLO_)md5.$(OBJ) $(C_) $(GLGEN)md5.c
 	$(RM_) $(GLGEN)md5.c $(GLGEN)md5.h
 
@@ -631,19 +631,19 @@ $(GLOBJ)gconfig.$(OBJ) : $(gconfig_h) $(GLSRC)gconf.c $(AK) $(gx_h)\
  $(gxdevice_h) $(gxiclass_h) $(gxiodev_h) $(gxiparam_h) $(TOP_MAKEFILES)\
  $(MAKEDDIRS)
 	$(RM_) $(GLGEN)gconfig.c
-	$(CP_) $(GLSRC)gconf.c $(GLGEN)gconfig.c
+	$(CP_) $(GLSRC)gconf.c $(GLGEN)gconfig.c || true
 	$(GLCC) $(GLO_)gconfig.$(OBJ) $(C_) $(GLGEN)gconfig.c
 
 $(GLOBJ)gscdefs.$(OBJ) : $(GLSRC)gscdef.c\
  $(std_h) $(gscdefs_h) $(gconfigd_h) $(TOP_MAKEFILES) $(MAKEDIRS)
 	$(RM_) $(GLGEN)gscdefs.c
-	$(CP_) $(GLSRC)gscdef.c $(GLGEN)gscdefs.c
+	$(CP_) $(GLSRC)gscdef.c $(GLGEN)gscdefs.c || true
 	$(GLCC) $(GLO_)gscdefs.$(OBJ) $(C_) $(GLGEN)gscdefs.c
 
 $(AUX)gscdefs.$(OBJ) : $(GLSRC)gscdef.c\
  $(std_h) $(gscdefs_h) $(gconfigd_h) $(TOP_MAKEFILES) $(MAKEDIRS)
 	$(RM_) $(AUX)gscdefs.c
-	$(CP_) $(GLSRC)gscdef.c $(AUX)gscdefs.c
+	$(CP_) $(GLSRC)gscdef.c $(AUX)gscdefs.c || true
 	$(GLCCAUX) $(C_) $(AUXO_)gscdefs.$(OBJ) $(AUX)gscdefs.c
 
 $(GLOBJ)gxacpath.$(OBJ) : $(GLSRC)gxacpath.c $(AK) $(gx_h)\
@@ -1535,7 +1535,7 @@ $(GLOBJ)sjpegc_0.$(OBJ) : $(GLSRC)sjpegc.c $(AK) $(stdio__h) $(string__h)\
 	$(GLJCC) $(GLO_)sjpegc_0.$(OBJ) $(C_) $(GLSRC)sjpegc.c
 
 $(GLOBJ)sjpegc.$(OBJ) : $(GLOBJ)sjpegc_$(SHARE_JPEG).$(OBJ)
-	$(CP_) $(GLOBJ)sjpegc_$(SHARE_JPEG).$(OBJ) $(GLOBJ)sjpegc.$(OBJ)
+	$(CP_) $(GLOBJ)sjpegc_$(SHARE_JPEG).$(OBJ) $(GLOBJ)sjpegc.$(OBJ) || true
 
 # sdcparam is used by the filter operator and the PS/PDF writer.
 # It is not included automatically in sdcte/d.
@@ -1563,7 +1563,7 @@ $(GLOBJ)sdcte_0.$(OBJ) : $(GLSRC)sdcte.c $(AK)\
 	$(GLJCC) $(GLO_)sdcte_0.$(OBJ) $(C_) $(GLSRC)sdcte.c
 
 $(GLOBJ)sdcte.$(OBJ) : $(GLOBJ)sdcte_$(SHARE_JPEG).$(OBJ) $(MAKEDIRS)
-	$(CP_) $(GLOBJ)sdcte_$(SHARE_JPEG).$(OBJ) $(GLOBJ)sdcte.$(OBJ)
+	$(CP_) $(GLOBJ)sdcte_$(SHARE_JPEG).$(OBJ) $(GLOBJ)sdcte.$(OBJ) || true
 
 
 $(GLOBJ)sjpege_1.$(OBJ) : $(GLSRC)sjpege.c $(AK)\
@@ -1579,7 +1579,7 @@ $(GLOBJ)sjpege_0.$(OBJ) : $(GLSRC)sjpege.c $(AK)\
 	$(GLJCC) $(GLO_)sjpege_0.$(OBJ) $(C_) $(GLSRC)sjpege.c
 
 $(GLOBJ)sjpege.$(OBJ) : $(GLOBJ)sjpege_$(SHARE_JPEG).$(OBJ) $(MAKEDIRS)
-	$(CP_) $(GLOBJ)sjpege_$(SHARE_JPEG).$(OBJ) $(GLOBJ)sjpege.$(OBJ)
+	$(CP_) $(GLOBJ)sjpege_$(SHARE_JPEG).$(OBJ) $(GLOBJ)sjpege.$(OBJ) || true
 
 # sdeparam is used by the filter operator and the PS/PDF writer.
 # It is not included automatically in sdcte.
@@ -1611,7 +1611,7 @@ $(GLOBJ)sdctd_0.$(OBJ) : $(GLSRC)sdctd.c $(AK)\
 	$(GLJCC) $(GLO_)sdctd_0.$(OBJ) $(C_) $(GLSRC)sdctd.c
 
 $(GLOBJ)sdctd.$(OBJ) : $(GLOBJ)sdctd_$(SHARE_JPEG).$(OBJ) $(MAKEDIRS)
-	$(CP_) $(GLOBJ)sdctd_$(SHARE_JPEG).$(OBJ) $(GLOBJ)sdctd.$(OBJ)
+	$(CP_) $(GLOBJ)sdctd_$(SHARE_JPEG).$(OBJ) $(GLOBJ)sdctd.$(OBJ) || true
 
 
 $(GLOBJ)sjpegd_1.$(OBJ) : $(GLSRC)sjpegd.c $(AK)\
@@ -1628,7 +1628,7 @@ $(GLOBJ)sjpegd_0.$(OBJ) : $(GLSRC)sjpegd.c $(AK)\
 
 
 $(GLOBJ)sjpegd.$(OBJ) : $(GLOBJ)sjpegd_$(SHARE_JPEG).$(OBJ) $(MAKEDIRS)
-	$(CP_) $(GLOBJ)sjpegd_$(SHARE_JPEG).$(OBJ) $(GLOBJ)sjpegd.$(OBJ)
+	$(CP_) $(GLOBJ)sjpegd_$(SHARE_JPEG).$(OBJ) $(GLOBJ)sjpegd.$(OBJ) || true
 
 # sddparam is used by the filter operator.
 # It is not included automatically in sdctd.
@@ -1651,7 +1651,7 @@ $(GLD)lzwe.dev : $(LIB_MAK) $(ECHOGS_XE) $(lzwe_)
 
 # We need slzwe.dev as a synonym for lzwe.dev for BAND_LIST_STORAGE = memory.
 $(GLD)slzwe.dev : $(GLD)lzwe.dev
-	$(CP_) $(GLD)lzwe.dev $(GLD)slzwe.dev
+	$(CP_) $(GLD)lzwe.dev $(GLD)slzwe.dev || true
 
 $(GLOBJ)slzwe.$(OBJ) : $(GLSRC)slzwe.c $(AK) $(stdio__h) $(gdebug_h)\
  $(slzwx_h) $(strimpl_h) $(MAKEDIRS)
@@ -1667,7 +1667,7 @@ $(GLD)lzwd.dev : $(LIB_MAK) $(ECHOGS_XE) $(lzwd_)
 
 # We need slzwd.dev as a synonym for lzwd.dev for BAND_LIST_STORAGE = memory.
 $(GLD)slzwd.dev : $(GLD)lzwd.dev
-	$(CP_) $(GLD)lzwd.dev $(GLD)slzwd.dev
+	$(CP_) $(GLD)lzwd.dev $(GLD)slzwd.dev || true
 
 $(GLOBJ)slzwd.$(OBJ) : $(GLSRC)slzwd.c $(AK) $(stdio__h) $(gdebug_h)\
  $(slzwx_h) $(strimpl_h) $(MAKEDIRS)
@@ -1717,7 +1717,7 @@ $(GLOBJ)saes.$(OBJ) : $(GLSRC)saes.c $(AK) $(memory__h)\
 # ---------------- JBIG2 compression filter ---------------- #
 
 $(GLD)sjbig2.dev : $(LIB_MAK) $(ECHOGS_XE) $(GLD)sjbig2_$(JBIG2_LIB).dev
-	$(CP_) $(GLD)sjbig2_$(JBIG2_LIB).dev $(GLD)sjbig2.dev
+	$(CP_) $(GLD)sjbig2_$(JBIG2_LIB).dev $(GLD)sjbig2.dev || true
 
 # jbig2dec version
 sjbig2_jbig2dec=$(GLOBJ)sjbig2.$(OBJ)
@@ -1753,7 +1753,7 @@ $(GLOBJ)sjbig2_luratech.$(OBJ) : $(GLSRC)sjbig2_luratech.c $(AK) \
 # ---------------- JPEG 2000 compression filter ---------------- #
 
 $(GLD)sjpx.dev : $(LIB_MAK) $(ECHOGS_XE) $(GLD)sjpx_$(JPX_LIB).dev
-	$(CP_) $(GLD)sjpx_$(JPX_LIB).dev $(GLD)sjpx.dev
+	$(CP_) $(GLD)sjpx_$(JPX_LIB).dev $(GLD)sjpx.dev || true
 
 $(GLOBJ)sjpx.$(OBJ) : $(GLSRC)sjpx.c $(AK) \
  $(memory__h) $(gsmalloc_h) \
@@ -1882,7 +1882,7 @@ $(GLOBJ)szlibc_0.$(OBJ) : $(GLSRC)szlibc.c $(AK) $(std_h)\
 	$(GLZCC) $(GLO_)szlibc_0.$(OBJ) $(C_) $(GLSRC)szlibc.c
 
 $(GLOBJ)szlibc.$(OBJ) : $(GLOBJ)szlibc_$(SHARE_ZLIB).$(OBJ) $(MAKEDIRS)
-	$(CP_) $(GLOBJ)szlibc_$(SHARE_ZLIB).$(OBJ) $(GLOBJ)szlibc.$(OBJ)
+	$(CP_) $(GLOBJ)szlibc_$(SHARE_ZLIB).$(OBJ) $(GLOBJ)szlibc.$(OBJ) || true
 
 szlibe_=$(szlibc_) $(GLOBJ)szlibe.$(OBJ)
 $(GLD)szlibe.dev : $(LIB_MAK) $(ECHOGS_XE) $(ZGENDIR)$(D)zlibe.dev $(szlibe_)
@@ -1898,7 +1898,7 @@ $(GLOBJ)szlibe_0.$(OBJ) : $(GLSRC)szlibe.c $(AK) $(std_h)\
 	$(GLZCC) $(GLO_)szlibe_0.$(OBJ) $(C_) $(GLSRC)szlibe.c
 
 $(GLOBJ)szlibe.$(OBJ) : $(GLOBJ)szlibe_$(SHARE_ZLIB).$(OBJ)  $(MAKEDIRS)
-	$(CP_) $(GLOBJ)szlibe_$(SHARE_ZLIB).$(OBJ) $(GLOBJ)szlibe.$(OBJ)
+	$(CP_) $(GLOBJ)szlibe_$(SHARE_ZLIB).$(OBJ) $(GLOBJ)szlibe.$(OBJ) || true
 
 szlibd_=$(szlibc_) $(GLOBJ)szlibd.$(OBJ)
 $(GLD)szlibd.dev : $(LIB_MAK) $(ECHOGS_XE) $(ZGENDIR)$(D)zlibd.dev $(szlibd_)
@@ -1914,7 +1914,7 @@ $(GLOBJ)szlibd_0.$(OBJ) : $(GLSRC)szlibd.c $(AK) $(std_h) $(memory__h)\
 	$(GLZCC) $(GLO_)szlibd_0.$(OBJ) $(C_) $(GLSRC)szlibd.c
 
 $(GLOBJ)szlibd.$(OBJ) : $(GLOBJ)szlibd_$(SHARE_ZLIB).$(OBJ) $(MAKEDIRS)
-	$(CP_) $(GLOBJ)szlibd_$(SHARE_ZLIB).$(OBJ) $(GLOBJ)szlibd.$(OBJ)
+	$(CP_) $(GLOBJ)szlibd_$(SHARE_ZLIB).$(OBJ) $(GLOBJ)szlibd.$(OBJ) || true
 
 # ---------------- Page devices ---------------- #
 # We include this here, rather than in devs.mak, because it is more like
@@ -2804,7 +2804,7 @@ $(GLOBJ)gsicc_lcms_0.$(OBJ) : $(GLSRC)gsicc_lcms.c\
 	$(GLLCMSCC) $(GLO_)gsicc_lcms_0.$(OBJ) $(C_) $(GLSRC)gsicc_lcms.c
 
 $(GLOBJ)gsicc_lcms.$(OBJ) : $(GLOBJ)gsicc_lcms_$(SHARE_LCMS).$(OBJ) $(gp_h)
-	$(CP_) $(GLOBJ)gsicc_lcms_$(SHARE_LCMS).$(OBJ) $(GLOBJ)gsicc_lcms.$(OBJ)
+	$(CP_) $(GLOBJ)gsicc_lcms_$(SHARE_LCMS).$(OBJ) $(GLOBJ)gsicc_lcms.$(OBJ) || true
 
 
 $(GLOBJ)gsicc_lcms2_1.$(OBJ) : $(GLSRC)gsicc_lcms2.c\
@@ -2817,7 +2817,7 @@ $(GLOBJ)gsicc_lcms2_0.$(OBJ) : $(GLSRC)gsicc_lcms2.c\
 
 $(GLOBJ)gsicc_lcms2.$(OBJ) : $(GLOBJ)gsicc_lcms2_$(SHARE_LCMS).$(OBJ) $(gp_h) \
  $(gxsync_h)
-	$(CP_) $(GLOBJ)gsicc_lcms2_$(SHARE_LCMS).$(OBJ) $(GLOBJ)gsicc_lcms2.$(OBJ)
+	$(CP_) $(GLOBJ)gsicc_lcms2_$(SHARE_LCMS).$(OBJ) $(GLOBJ)gsicc_lcms2.$(OBJ) || true
 
 # Note that gsicc_create requires compile with lcms to obtain icc34.h
 # header file that is used for creating ICC structures from PS objects.
@@ -2837,7 +2837,7 @@ $(GLOBJ)gsicc_create_0.$(OBJ) : $(GLSRC)gsicc_create.c $(AK) $(string__h)\
 	$(GLLCMSCC) $(GLO_)gsicc_create_0.$(OBJ) $(C_) $(GLSRC)gsicc_create.c
 
 $(GLOBJ)gsicc_create.$(OBJ) : $(GLOBJ)gsicc_create_$(SHARE_LCMS).$(OBJ) $(MAKEDIRS)
-	$(CP_) $(GLOBJ)gsicc_create_$(SHARE_LCMS).$(OBJ) $(GLOBJ)gsicc_create.$(OBJ)
+	$(CP_) $(GLOBJ)gsicc_create_$(SHARE_LCMS).$(OBJ) $(GLOBJ)gsicc_create.$(OBJ) || true
 
 
 #include "icc34.h"   /* Note this header is needed even if lcms is not compiled as default CMS */
@@ -3141,7 +3141,7 @@ $(GLGEN)gsromfs1_1.c : $(MKROMFS_XE) $(PS_ROMFS_DEPS) $(MAKEDIRS)
 	$(PS_ROMFS_ARGS) $(GL_ROMFS_ARGS)
 
 $(GLGEN)gsromfs1.c : $(GLGEN)gsromfs1_$(UFST_BRIDGE).c $(MAKEDIRS)
-	$(CP_) $(GLGEN)gsromfs1_$(UFST_BRIDGE).c $(GLGEN)gsromfs1.c
+	$(CP_) $(GLGEN)gsromfs1_$(UFST_BRIDGE).c $(GLGEN)gsromfs1.c || true
 
 # the following module is only included if the romfs.dev FEATURE is enabled
 $(GLOBJ)gsiorom_1.$(OBJ) : $(GLSRC)gsiorom.c $(gsiorom_h) \
@@ -3157,7 +3157,7 @@ $(GLOBJ)gsiorom_0.$(OBJ) : $(GLSRC)gsiorom.c $(gsiorom_h) \
 	$(GLCC) $(GLO_)gsiorom_0.$(OBJ) $(I_)$(ZI_)$(_I) $(C_) $(GLSRC)gsiorom.c
 
 $(GLOBJ)gsiorom.$(OBJ) : $(GLOBJ)gsiorom_$(SHARE_ZLIB).$(OBJ) $(MAKEDIRS)
-	$(CP_) $(GLOBJ)gsiorom_$(SHARE_ZLIB).$(OBJ) $(GLOBJ)gsiorom.$(OBJ)
+	$(CP_) $(GLOBJ)gsiorom_$(SHARE_ZLIB).$(OBJ) $(GLOBJ)gsiorom.$(OBJ) || true
 
 $(GLOBJ)gsromfs1.$(OBJ) : $(GLOBJ)gsromfs1.c $(time__h) $(MAKEDIRS)
 	$(GLCC) $(GLO_)gsromfs1.$(OBJ) $(C_) $(GLOBJ)gsromfs1.c
diff --git a/base/lwf_jp2.mak b/base/lwf_jp2.mak
index ca1b7cc..ad6446d 100644
--- a/base/lwf_jp2.mak
+++ b/base/lwf_jp2.mak
@@ -176,7 +176,7 @@ lwf_jp2_HDRS = \
 
 # switch in the selected library .dev
 $(LWF_JP2_GEN)lwf_jp2.dev : $(TOP_MAKEFILES) $(LWF_JP2_GEN)lwf_jp2_$(SHARE_JPX).dev
-	$(CP_) $(LWF_JP2_GEN)lwf_jp2_$(SHARE_JPX).dev $(LWF_JP2_GEN)lwf_jp2.dev
+	$(CP_) $(LWF_JP2_GEN)lwf_jp2_$(SHARE_JPX).dev $(LWF_JP2_GEN)lwf_jp2.dev || true
 
 # external link .dev
 $(LWF_JP2_GEN)lwf_jp2_1.dev : $(TOP_MAKEFILES) $(LWF_JP2_MAK) $(ECHOGS_XE)
diff --git a/base/macos-mcp.mak b/base/macos-mcp.mak
index 38a2937..2ab7cae 100644
--- a/base/macos-mcp.mak
+++ b/base/macos-mcp.mak
@@ -383,11 +383,11 @@ CWPROJ_XML=./ghostscript.mcp.xml
 
 $(CWPROJ_XML): $(gconfigd_h)
 	-mkdir -p obj/sys
-	$(CP_) $(macsystypes_h) $(systypes_h)
+	$(CP_) $(macsystypes_h) $(systypes_h) || true
 	$(SH) $(GLSRC)macgenmcpxml.sh `$(CAT) $(ld_tr)` >  $(CWPROJ_XML)
-	$(CP_) $(GLSRC)gconf.c $(GLOBJ)gconfig.c
-	$(CP_) $(GLSRC)iconf.c $(GLOBJ)iconfig.c
-	$(CP_) $(GLSRC)gscdef.c $(GLOBJ)gscdefs.c
+	$(CP_) $(GLSRC)gconf.c $(GLOBJ)gconfig.c || true
+	$(CP_) $(GLSRC)iconf.c $(GLOBJ)iconfig.c || true
+	$(CP_) $(GLSRC)gscdef.c $(GLOBJ)gscdefs.c || true
 	/Developer/Tools/SetFile -c CWIE -t TEXT $(CWPROJ_XML)
 
 $(GS_XE): $(ld_tr) $(ECHOGS_XE) $(XE_ALL) $(CWPROJ_XML) $(PSOBJ)gsromfs$(COMPILE_INITS).$(OBJ)
diff --git a/base/openjpeg.mak b/base/openjpeg.mak
index b194709..dac7901 100644
--- a/base/openjpeg.mak
+++ b/base/openjpeg.mak
@@ -99,7 +99,7 @@ open_jpeg_HDRS = \
 
 # switch in the selected library .dev
 $(OPEN_JPEG_GEN)openjpeg.dev : $(TOP_MAKEFILES) $(OPEN_JPEG_GEN)openjpeg_$(SHARE_JPX).dev
-	$(CP_) $(OPEN_JPEG_GEN)openjpeg_$(SHARE_JPX).dev $(OPEN_JPEG_GEN)openjpeg.dev
+	$(CP_) $(OPEN_JPEG_GEN)openjpeg_$(SHARE_JPX).dev $(OPEN_JPEG_GEN)openjpeg.dev || true
 
 # external link .dev
 $(OPEN_JPEG_GEN)openjpeg_1.dev : $(TOP_MAKEFILES) $(OPEN_JPEG_MAK) $(ECHOGS_XE)
diff --git a/base/openvms.mak b/base/openvms.mak
index 1ea06fe..84378a4 100644
--- a/base/openvms.mak
+++ b/base/openvms.mak
@@ -365,7 +365,7 @@ SH=
 
 # Define generic commands.
 
-CP_=$$ @$(GLSRCDIR)COPY_ONE
+CP_=$$ @$(GLSRCDIR)COPY_ONE || true
 
 # Define the command for deleting (a) file(s) (including wild cards)
 
diff --git a/base/png.mak b/base/png.mak
index 1725ddb..0d54b80 100644
--- a/base/png.mak
+++ b/base/png.mak
@@ -79,7 +79,7 @@ png.config-clean :
 	$(RM_) $(PNGGEN)lpg*.dev
 
 $(pnglibconf_h) : $(PNGSRC)scripts$(D)pnglibconf.h.prebuilt
-	$(CP_)  $(PNGSRC)scripts$(D)pnglibconf.h.prebuilt $(pnglibconf_h)
+	$(CP_)  $(PNGSRC)scripts$(D)pnglibconf.h.prebuilt $(pnglibconf_h) || true
 
 PDEP=$(AK) $(pnglibconf_h)
 
@@ -134,7 +134,7 @@ $(PNGOBJ)pngget.$(OBJ) : $(PNGSRC)pngget.c $(PDEP)
 
 # Define the version of libpng.dev that we are actually using.
 $(PNGGEN)libpng.dev : $(TOP_MAKEFILES) $(PNGGEN)libpng_$(SHARE_LIBPNG).dev $(MAKEDIRS)
-	$(CP_) $(PNGGEN)libpng_$(SHARE_LIBPNG).dev $(PNGGEN)libpng.dev
+	$(CP_) $(PNGGEN)libpng_$(SHARE_LIBPNG).dev $(PNGGEN)libpng.dev || true
 
 # Define the shared version of libpng.
 # Note that it requires libz, which must be searched *after* libpng.
diff --git a/base/tiff.mak b/base/tiff.mak
index 0ea9f9e..80d8926 100644
--- a/base/tiff.mak
+++ b/base/tiff.mak
@@ -53,11 +53,11 @@ tiff_11=$(TIFFOBJ)gstiffio.$(OBJ)
 
 $(TIFFSRC)libtiff$(D)tif_config.unix.h : $(TIFFSRC)libtiff$(D)tif_config.h.in
 	cd $(TIFFSRC) && ./configure
-	$(CP_) $(TIFFCONF)libtiff$(D)tif_config.h $(TIFFCONF)libtiff$(D)tif_config.unix.h
+	$(CP_) $(TIFFCONF)libtiff$(D)tif_config.h $(TIFFCONF)libtiff$(D)tif_config.unix.h || true
 
 $(TIFFSRC)libtiff$(D)tiffconf.unix.h : $(TIFFSRC)libtiff$(D)tiffconf.h.in
 	cd $(TIFFSRC) && ./configure
-	$(CP_) $(TIFFCONF)libtiff$(D)tiffconf.h $(TIFFCONF)libtiff$(D)tiffconf.unix.h
+	$(CP_) $(TIFFCONF)libtiff$(D)tiffconf.h $(TIFFCONF)libtiff$(D)tiffconf.unix.h || true
 
 $(TIFFOBJ)tif_aux.$(OBJ) : $(TIFFSRC)/libtiff/tif_aux.c $(TIFFDEP)
 	$(TIFFCC) $(TIFFO_)tif_aux.$(OBJ) $(C_) $(TIFFSRC)/libtiff/tif_aux.c
@@ -189,14 +189,14 @@ $(TIFFOBJ)gstiffio.$(OBJ) : $(GLSRC)gstiffio.c $(gstiffio_h) $(PDEVH) $(stdint__
 	$(TIFFCC) $(TIFFO_)gstiffio.$(OBJ) $(D_)SHARE_LIBTIFF=$(SHARE_LIBTIFF) $(C_) $(GLSRC)gstiffio.c
 
 $(TIFFGEN)tif_config.h: $(TIFFCONFIG_H)
-	$(CP_) $(TIFFCONFIG_H) $(TIFFGEN)tif_config.h
+	$(CP_) $(TIFFCONFIG_H) $(TIFFGEN)tif_config.h || true
 	
 $(TIFFGEN)tiffconf.h: $(TIFFCONF_H)
-	$(CP_) $(TIFFCONF_H) $(TIFFGEN)tiffconf.h
+	$(CP_) $(TIFFCONF_H) $(TIFFGEN)tiffconf.h || true
 	
 # Define the version of libtiff.dev that we are actually using.
 $(TIFFGEN)libtiff.dev : $(TOP_MAKEFILES) $(TIFFGEN)libtiff_$(SHARE_LIBTIFF).dev
-	$(CP_) $(TIFFGEN)libtiff_$(SHARE_LIBTIFF).dev $(TIFFGEN)libtiff.dev
+	$(CP_) $(TIFFGEN)libtiff_$(SHARE_LIBTIFF).dev $(TIFFGEN)libtiff.dev || true
 
 
 # Define the shared version.
diff --git a/base/unix-aux.mak b/base/unix-aux.mak
index 1ddad78..6b24422 100644
--- a/base/unix-aux.mak
+++ b/base/unix-aux.mak
@@ -105,7 +105,7 @@ $(GLOBJ)gp_sysv.$(OBJ): $(GLSRC)gp_sysv.c $(stdio__h) $(time__h) $(AK)\
 #	$(CCAUX_) $(GENOPT) $(CFLAGS) $(I_)$(GLSRCDIR)$(_I) $(I_)$(GLOBJ)$(_I) $(I_)$(ZSRCDIR)$(_I) $(GLSRC)mkromfs.c $(O_)$(MKROMFS_XE)_1 $(MKROMFS_OBJS_1) $(AUXEXTRALIBS)
 #
 #$(MKROMFS_XE): $(MKROMFS_XE)_$(SHARE_ZLIB) $(MAKEDIRS)
-#	$(CP_) $(MKROMFS_XE)_$(SHARE_ZLIB) $(MKROMFS_XE)
+#	$(CP_) $(MKROMFS_XE)_$(SHARE_ZLIB) $(MKROMFS_XE) || true
 
 # Query the environment to construct gconfig_.h.
 # These are all defined conditionally (except the JasPER one), so that
diff --git a/base/unixhead.mak b/base/unixhead.mak
index 6aa3599..70a61f8 100644
--- a/base/unixhead.mak
+++ b/base/unixhead.mak
@@ -53,7 +53,7 @@ SH=$(SHELL)
 
 # Define generic commands.
 
-CP_=cp
+CP_=cp -f
 RM_=rm -f
 RMN_=rm -f
 
diff --git a/base/zlib.mak b/base/zlib.mak
index 78d398a..3d48145 100644
--- a/base/zlib.mak
+++ b/base/zlib.mak
@@ -83,7 +83,7 @@ $(ZOBJ)zutil.$(OBJ) : $(ZSRC)zutil.c $(ZDEP)
 # Encoding (compression) code.
 
 $(ZGEN)zlibe.dev : $(TOP_MAKEFILES) $(ZGEN)zlibe_$(SHARE_ZLIB).dev
-	$(CP_) $(ZGEN)zlibe_$(SHARE_ZLIB).dev $(ZGEN)zlibe.dev
+	$(CP_) $(ZGEN)zlibe_$(SHARE_ZLIB).dev $(ZGEN)zlibe.dev || true
 
 $(ZGEN)zlibe_1.dev : $(TOP_MAKEFILES) $(ZLIB_MAK) $(ECHOGS_XE)
 	$(SETMOD) $(ZGEN)zlibe_1 -lib $(ZLIB_NAME)
@@ -111,7 +111,7 @@ $(ZOBJ)trees.$(OBJ) : $(ZSRC)trees.c $(ZDEP)
 # with 0.90 do.
 
 $(ZGEN)crc32.dev : $(TOP_MAKEFILES) $(ZGEN)crc32_$(SHARE_ZLIB).dev
-	$(CP_) $(ZGEN)crc32_$(SHARE_ZLIB).dev $(ZGEN)crc32.dev
+	$(CP_) $(ZGEN)crc32_$(SHARE_ZLIB).dev $(ZGEN)crc32.dev || true
 
 $(ZGEN)crc32_1.dev : $(TOP_MAKEFILES) $(ZLIB_MAK) $(ECHOGS_XE)
 	$(SETMOD) $(ZGEN)crc32_1 -lib $(ZLIB_NAME)
@@ -127,7 +127,7 @@ $(ZOBJ)crc32.$(OBJ) : $(ZSRC)crc32.c $(ZDEP)
 # Decoding (decompression) code.
 
 $(ZGEN)zlibd.dev : $(TOP_MAKEFILES) $(ZGEN)zlibd_$(SHARE_ZLIB).dev $(MAKEDIRS)
-	$(CP_) $(ZGEN)zlibd_$(SHARE_ZLIB).dev $(ZGEN)zlibd.dev
+	$(CP_) $(ZGEN)zlibd_$(SHARE_ZLIB).dev $(ZGEN)zlibd.dev || true
 
 $(ZGEN)zlibd_1.dev : $(TOP_MAKEFILES) $(ZLIB_MAK) $(ECHOGS_XE)
 	$(SETMOD) $(ZGEN)zlibd_1 -lib $(ZLIB_NAME)
diff --git a/devices/devs.mak b/devices/devs.mak
index 0a668d4..a19082e 100644
--- a/devices/devs.mak
+++ b/devices/devs.mak
@@ -1547,7 +1547,7 @@ libpng_dev=$(PNGGENDIR)$(D)libpng.dev
 png_i_=-include $(PNGGENDIR)$(D)libpng
 
 $(DEVOBJ)gdevpng.$(OBJ) : $(DEVSRC)gdevpng.c\
- $(gdevprn_h) $(gdevpccm_h) $(gscdefs_h) $(png__h)
+ $(gdevprn_h) $(gdevpccm_h) $(gscdefs_h) $(png__h) $(PDEP)
 	$(CC_) $(I_)$(DEVI_) $(II)$(PI_)$(_I) $(PCF_) $(GLF_) $(DEVO_)gdevpng.$(OBJ) $(C_) $(DEVSRC)gdevpng.c
 
 $(DD)pngmono.dev : $(DEVS_MAK) $(libpng_dev) $(png_) $(GLD)page.dev $(GDEV)
diff --git a/psi/int.mak b/psi/int.mak
index 7d36e66..71b88f5 100644
--- a/psi/int.mak
+++ b/psi/int.mak
@@ -274,7 +274,7 @@ $(PSOBJ)iconfig.$(OBJ) : $(gconfig_h) $(PSSRC)iconf.c $(stdio__h)\
  $(gconf_h) $(gconfigd_h) $(gsmemory_h) $(gstypes_h)\
  $(iminst_h) $(iref_h) $(ivmspace_h) $(opdef_h) $(iplugin_h)
 	$(RM_) $(PSGEN)iconfig.c
-	$(CP_) $(PSSRC)iconf.c $(PSGEN)iconfig.c
+	$(CP_) $(PSSRC)iconf.c $(PSGEN)iconfig.c || true
 	$(PSCC) $(PSO_)iconfig.$(OBJ) $(C_) $(PSGEN)iconfig.c
 
 $(PSOBJ)iinit.$(OBJ) : $(PSSRC)iinit.c $(GH) $(string__h)\
@@ -607,7 +607,7 @@ $(PSD)psbase.dev : $(INT_MAK) $(ECHOGS_XE) $(INT_OBJS)\
 
 # We keep the old name for backward compatibility.
 $(PSD)level1.dev : $(PSD)psl1.dev
-	$(CP_) $(PSD)psl1.dev $(PSD)level1.dev
+	$(CP_) $(PSD)psl1.dev $(PSD)level1.dev || true
 
 $(PSD)psl1.dev : $(INT_MAK) $(ECHOGS_XE)\
  $(PSD)psbase.dev $(PSD)bcp.dev $(PSD)path1.dev $(PSD)type1.dev
@@ -934,7 +934,7 @@ $(PSOBJ)zfrsd.$(OBJ) : $(PSSRC)zfrsd.c $(OP) $(memory__h)\
 
 # We keep the old name for backward compatibility.
 $(PSD)level2.dev : $(PSD)psl2.dev
-	$(CP_) $(PSD)psl2.dev $(PSD)level2.dev
+	$(CP_) $(PSD)psl2.dev $(PSD)level2.dev || true
 
 # We -include dpsand2 first so that geninit will have access to the
 # system name table as soon as possible.
@@ -1193,7 +1193,7 @@ $(PSOBJ)zfjbig2_luratech.$(OBJ) : $(PSSRC)zfjbig2.c $(OP) $(memory__h)\
 # this can be turned on and off with a FEATURE_DEV
 
 $(PSD)jpx.dev : $(INT_MAK) $(ECHOGS_XE) $(PSD)jpx_$(JPX_LIB).dev
-	$(CP_) $(PSD)jpx_$(JPX_LIB).dev $(PSD)jpx.dev
+	$(CP_) $(PSD)jpx_$(JPX_LIB).dev $(PSD)jpx.dev || true
 
 fjpx_luratech=$(PSOBJ)zfjpx_luratech.$(OBJ)
 
-- 
1.9.1

